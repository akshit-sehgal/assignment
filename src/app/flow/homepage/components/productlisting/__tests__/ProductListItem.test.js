import React from 'react';
import { shallow } from 'enzyme';
import ProductListingItem from '../ProductListItem';
import { mockProductListingData } from '../__mocks__/productListing.mock';

describe('Testing <ProductListItem />', () => {
    let props;
    beforeEach(() => {
        const { id, name, img, price } = mockProductListingData.productsList[0];
        props = {
            id,
            name,
            img,
            price,
            navigate: jest.fn()
        }
    });
    it('should render the default view', () => {
        const wrapper = shallow(<ProductListingItem {...props} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.pli__img').prop('src')).toEqual(props.img);
        expect(wrapper.find('.pli__name').prop('children')).toEqual(props.name);
        expect(wrapper.find('.pli__price').prop('children')).toEqual(`Rs. ${props.price} /-`);
    });
    it('should call navigate', () => {
        const wrapper = shallow(<ProductListingItem {...props} />);
        wrapper.find('.product-list-item').simulate('click');
        expect(props.navigate).toHaveBeenCalledTimes(1);
    });

});