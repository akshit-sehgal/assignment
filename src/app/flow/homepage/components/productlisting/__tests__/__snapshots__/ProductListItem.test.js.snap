// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ProductListItem /> should render the default view 1`] = `
<div
  className="product-list-item"
  data-id={41}
  onClick={[MockFunction]}
>
  <img
    className="pli__img"
    src="https://ecs7.tokopedia.net/img/cache/200-square/product-1/2019/6/12/2570456/2570456_90794ecf-e3f4-4896-b466-efe108372d5a_720_720.jpg.webp"
  />
  <div
    className="pli__description"
  >
    <div
      className="pli__name"
    >
      Product 41
    </div>
    <div
      className="pli__price"
    >
      Rs. 1000 /-
    </div>
  </div>
</div>
`;
